src/views/text_view.rs

```diff
@@ -12,6 +14,9 @@ use crate::utils::markup::StyledString;
 use crate::view::{SizeCache, View};
 use crate::{Printer, Vec2, With, XY};
 
+// Content type used internally for caching and storage
+type InnerContentType = RefCell<Arc<StyledString>>;
```

Comment:
Do you need a `RefCell` here? `TextContentInner` is behind a `Mutex`, so anytime it is accessed it should mutable; we probably don't need this extra interior mutability.

In addition, `RefCell` is `!Sync` so even with `Arc`, it probably won't work well with threads.
Note: we probably need some test to assert than `TextContent` is `Send`.